src/test/java/me/karakelley/http/HttpServerTest.java

```diff
@@ -0,0 +1,203 @@
+package me.karakelley.http;
+
+import ch.qos.logback.classic.Logger;
+import me.karakelley.http.helpers.ClientHelper;
+import me.karakelley.http.utility.Appender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.net.ConnectException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class HttpServerTest {
+  private ClientHelper client;
+  private HttpServer httpServer;
+  private HttpServer httpServerOnSamePort;
+  int count;
+  int maxTries;
+
+  private static Logger rootLogger;
+  private static Appender appender;
+
+  @BeforeEach
+  void setUp() {
+    rootLogger = (Logger) LoggerFactory.getLogger("ROOT");
+    appender = (Appender) rootLogger.getAppender("appender");
+    appender.setPrefix("test");
+    appender.start();
+    client = new ClientHelper();
+    count = 0;
+    maxTries = 4;
+  }
+
+  @AfterEach
+  void tearDown() {
+    appender.stop();
+  }
+
+  @Test
+  void testSendsResponseGivenValidRequest() throws Exception {
+    httpServer = new HttpServer(0);
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    executorService.submit(() -> httpServer.start());
+
+    try {
+      Thread.sleep(10);
+      client.connect("127.0.0.1", httpServer.getPortNumber());
+    } catch (ConnectException ex) {
+      count++;
+      if (count > maxTries) {
+        throw ex;
+      }
+    }
```

Comment:
I still feel like there's a lot of duplication in these tests. You've got a nice implementation of `ClientHelper` going here. Maybe there's an opportunity to turn your `connect` method into a `connectWithRetry`.

Reply:
Oh that's a great idea!